File: RiceCooker.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

void afficherMenu() {
    printf("X--------X-------X\n");
    printf("Que souhaitez-vous faire?\n");
    printf("X--------X-------X\n");
    printf("1. Cuire du riz\n");
    printf("2. Cuisson à la vapeur\n");
    printf("3. Cuisson des céréales\n");
    printf("4. Cuisson de la soupe\n");
    printf("5. Bouillir de l'eau\n");
    printf("6. Cuisson du dessert\n");
    printf("7. Maintenir au chaud\n");
    printf("8. Quitter\n");
}

int choisirOption() {
    int option;
    printf("Choisissez une option : ");
    scanf("%d", &option);
    if (option < 1 || option > 8) {
        printf("Option invalide. Veuillez choisir un nombre entre 1 et 8.\n");
        exit(1);
    }
    return option;
}

void attendre(int ms) {
    usleep(ms * 1000); // Attendre en millisecondes
}

int main() {
    int option;
    while (1) {
        afficherMenu();
        option = choisirOption();
        switch (option) {
            case 1:
                printf("Cuisson du riz en cours...\n");
                attendre(5000); 
                printf("Le riz est cuit\n");
                attendre(5000); 
                printf("Cuisson terminée\n");
                break;
            case 2:
                printf("Cuisson à la vapeur\n");
                attendre(5000); 
                printf("Veuillez Patientez SVP\n");
                attendre(5000); 
                printf("Cuisson terminée\n");
                break;
            case 3:
                printf("Cuisson des céréales\n");
                attendre(5000); 
                printf("Cuisson terminée\n");
                attendre(5000); 
                printf("Ca y est c'est prêt\n");
                break;
            case 4:
                printf("Cuisson de la soupe\n");
                attendre(5000); 
                printf("Cuisson terminée\n");
                attendre(5000); 
                printf("La soupe est prête\n");
                break;
            case 5:
                printf("Bouillir de l'eau\n");
                attendre(5000); 
                printf("tic tic tic tic\n");
                attendre(5000); 
                printf("L'eau est chaude\n");
                break;
            case 6:
                printf("Cuisson du dessert...\n");
                attendre(5000); 
                printf("Cuisson terminée\n");
                attendre(5000); 
                printf("C'est prêt\n");
                break;
            case 7:
                printf("Maintenir au chaud\n");
                attendre(5000); 
                break;
            case 8:
                printf("Au revoir et bon appétit !!!\n");
                return 0;
            default:
                printf("Option non reconnue\n");
                break;
        }
    }
    return 0;
}

File: RiceCooker.rb
# frozen_string_literal: true

def afficher_menu
  puts 'Que souhaitez-vous faire?'
  puts '1. Cuire du riz'
  puts '2. Cuisson à la vapeur'
  puts '3. Cuisson des céréales'
  puts '4. Cuisson de la soupe'
  puts '5. Bouillir de l\'eau'
  puts '6. Cuisson du dessert'
  puts '7. Maintenir au chaud' 
  puts '8. Quitter'
end

def choisir_option
  print 'Choisissez une option : '
  choix = gets.chomp.to_i
  raise 'Option invalide. Veuillez choisir un nombre entre 1 et 8.' unless (1..8).include?(choix)

  choix
end

def attendre(msi)
  sleep(msi / 1000.0)
end

def cuire_riz
  puts 'Cuisson du riz en cours...'
  attendre(5000) # Attendre 5 secondes
  puts 'Le riz est cuit'
  attendre(5000) # Attendre 5 secondes
  puts 'Cuisson terminée'
end

def cuisson_vapeur
  puts 'Cuisson à la vapeur...'
  attendre(5000) # Attendre 5 secondes
  puts 'Veuillez Patientez SVP'
  attendre(5000) # Attendre 5 secondes
  puts 'Cuisson terminée'
end

def cuisson_cereales
  puts 'Cuisson des céréales...'
  attendre(5000) # Attendre 5 secondes
  puts 'Cuisson terminée'
  attendre(5000) # Attendre 5 secondes
  puts 'Ca y est c\'est prêt'
end

def cuisson_soupe
  puts 'Cuisson de la soupe...'
  attendre(5000) # Attendre 5 secondes
  puts 'Cuisson terminée'
  attendre(5000) # Attendre 5 secondes
  puts 'La soupe est prête'
end

def bouillir_eau
  puts 'Bouillir de l\'eau...'
  attendre(5000) # Attendre 5 secondes
  puts 'tic tic tic tic'
  attendre(5000) # Attendre 5 secondes
  puts 'L\'eau est chaude'
end

def cuisson_dessert
  puts 'Cuisson du dessert...'
  attendre(5000) # Attendre 5 secondes
  puts 'Cuisson terminée'
  attendre(5000) # Attendre 5 secondes
  puts 'C\'est prêt'
end

def maintenir_chaud
  puts 'Maintenir au chaud...'
  attendre(5000) # Attendre 5 secondes
end

def main
  loop do
    afficher_menu
    begin
      option = choisir_option
      case option
      when 1
        cuire_riz
      when 2
        cuisson_vapeur
      when 3
        cuisson_cereales
      when 4
        cuisson_soupe
      when 5
        bouillir_eau
      when 6
        cuisson_dessert
      when 7
        maintenir_chaud
      when 8
        puts 'Au revoir et bon appétit !!!'
        return
      end
    rescue StandardError => e
      puts e
    end
  end
end

main
